File: examples/bench/fs_list.py
import os
import sys
import time

root = sys.argv[1] if len(sys.argv) > 1 else "bench_tree"
t0 = time.perf_counter()
count = 0
for name in os.listdir(root):
    full = os.path.join(root, name)
    if os.path.isdir(full):
        count += 1
        for child in os.listdir(full):
            cfull = os.path.join(full, child)
            if os.path.isfile(cfull):
                count += 1
    elif os.path.isfile(full):
        count += 1
t1 = time.perf_counter()
print("INNER_MS=" + str(int(round((t1 - t0) * 1000))))
print("COUNT=" + str(count))

File: examples/bench/fs_list.js
'use strict';
const fs = require('fs');
const path = require('path');
const { performance } = require('perf_hooks');

const root = process.argv[2] || 'bench_tree';
const t0 = performance.now();
let count = 0;
for (const name of fs.readdirSync(root)) {
  const full = path.join(root, name);
  const st = fs.statSync(full);
  if (st.isDirectory()) {
    count++;
    for (const child of fs.readdirSync(full)) {
      const cfull = path.join(full, child);
      if (fs.statSync(cfull).isFile()) count++;
    }
  } else if (st.isFile()) {
    count++;
  }
}
const t1 = performance.now();
console.log('INNER_MS=' + Math.round(t1 - t0));
console.log('COUNT=' + count);
